server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -325,6 +438,83 @@ public RecoverySettings(Settings settings, ClusterSettings clusterSettings) {
         );
     }
 
+    private void computeMaxBytesPerSec(Settings settings) {
+        validateBandwidthsSettings(settings);
+
+        // limit as computed before 8.1.0
+        final long defaultBytesPerSec = Math.max(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.get(settings).getBytes(), 0L);
+
+        // available network bandwidth
+        final long networkBandwidthBytesPerSec = Math.max(availableNetworkBandwidth.getBytes(), 0L);
+
+        // available disk bandwidth
+        final long diskBandwidthBytesPerSec;
+        if (availableDiskReadBandwidth.getBytes() > 0L && availableDiskWriteBandwidth.getBytes() > 0L) {
+            final long diskReadBandwidth = Math.round(
+                availableDiskReadBandwidth.getBytes() * MAX_BYTES_PER_SEC_USER_FACTOR_READS_SETTING.get(settings)
+            );
+            final long diskWriteBandwidth = Math.round(
+                availableDiskWriteBandwidth.getBytes() * MAX_BYTES_PER_SEC_USER_FACTOR_WRITES_SETTING.get(settings)
+            );
+            diskBandwidthBytesPerSec = Math.min(diskReadBandwidth, diskWriteBandwidth);
+        } else {
+            diskBandwidthBytesPerSec = 0L;
+        }
+
+        final long availableBytesPerSec = Math.round(
+            Math.min(diskBandwidthBytesPerSec, networkBandwidthBytesPerSec) * MAX_BYTES_PER_SEC_USER_FACTOR_SETTING.get(settings)
+        );
+
+        long maxBytesPerSec;
+        if (availableBytesPerSec == 0L                                      // no available bandwidths
+            || INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.exists(settings)  // when set this setting overrides available bandwidths
+            || DiscoveryNode.canContainData(settings) == false) {           // keep previous behavior for non data nodes
+            maxBytesPerSec = defaultBytesPerSec;
+        } else {
+            maxBytesPerSec = Math.max(defaultBytesPerSec, availableBytesPerSec);
+        }
+
+        final long maxAllowedBytesPerSec = Math.round(
+            Math.min(diskBandwidthBytesPerSec, networkBandwidthBytesPerSec) * MAX_BYTES_PER_SEC_OPERATOR_MAX_OVERCOMMIT_SETTING.get(
+                settings
+            )
+        );
+
+        long finalMaxBytesPerSec;
+        if (maxAllowedBytesPerSec > 0L) {
+            finalMaxBytesPerSec = Math.min(maxBytesPerSec, maxAllowedBytesPerSec);
+        } else {
+            finalMaxBytesPerSec = maxBytesPerSec;
+        }
+        logger.fatal("{} ad {}", MAX_BYTES_PER_SEC_USER_FACTOR_SETTING.get(settings), NODE_ROLES_SETTING.get(settings));
```

Comment:
Leftover?

Reply:
Yes, I removed this c0ff25d3f63b2291e743ee2ef37fbe3b93048053. Thanks.